src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvOrganizationServiceEntityUpdater.java

```diff
@@ -0,0 +1,149 @@
+import fi.otavanopisto.kuntaapi.server.controllers.IdentifierRelationController;
+import fi.otavanopisto.kuntaapi.server.discover.EntityUpdater;
+import fi.otavanopisto.kuntaapi.server.id.IdController;
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.id.OrganizationServiceId;
+import fi.otavanopisto.kuntaapi.server.id.ServiceId;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.cache.PtvOrganizationServiceCache;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.tasks.OrganizationServiceIdTaskQueue;
+import fi.otavanopisto.kuntaapi.server.persistence.model.Identifier;
+import fi.otavanopisto.kuntaapi.server.settings.SystemSettingController;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask.Operation;
+import fi.otavanopisto.restfulptv.client.ApiResponse;
+
+@ApplicationScoped
+@Singleton
+@AccessTimeout (unit = TimeUnit.HOURS, value = 1l)
+@SuppressWarnings ("squid:S3306")
+public class PtvOrganizationServiceEntityUpdater extends EntityUpdater {
+
+  private static final int TIMER_INTERVAL = 5000;
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private SystemSettingController systemSettingController;
+
+  @Inject
+  private PtvApi ptvApi;
+  
+  @Inject
+  private IdController idController;
+  
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private IdentifierRelationController identifierRelationController;
+
+  @Inject
+  private ModificationHashCache modificationHashCache;
+  
+  @Inject
+  private OrganizationServiceIdTaskQueue organizationServiceIdTaskQueue;
+  
+  @Inject
+  private PtvOrganizationServiceCache organizationServiceCache;
+  
+  @Inject
+  private PtvTranslator ptvTranslator;
+  
+  @Resource
+  private TimerService timerService;
+  
+  @Override
+  public String getName() {
+    return "organization-services";
+  }
+
+  @Override
+  public void startTimer() {
+    startTimer(TIMER_INTERVAL);
+  }
+
+  private void startTimer(int duration) {
+    TimerConfig timerConfig = new TimerConfig();
+    timerConfig.setPersistent(false);
+    timerService.createSingleActionTimer(duration, timerConfig);
+  }
+
+  @Timeout
+  public void timeout(Timer timer) {
+    executeNextTask();
+    startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
+  }
+  
+  private void executeNextTask() {
+    IdTask<OrganizationServiceId> task = organizationServiceIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateOrganizationService(task.getId(), task.getOrderIndex());
+      } else if (task.getOperation() == Operation.REMOVE) {
+        logger.log(Level.SEVERE, "PTV Organization service removal is not implemented");
+      }
+    }
+  }
+  
+  private void updateOrganizationService(OrganizationServiceId ptvOrganizationServiceId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateOrganizationService" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)